src/System.Text.Json/tests/Serialization/CustomConverterTests.Object.cs

```diff
@@ -139,5 +139,108 @@ public static void CustomObjectBoolConverter()
+
+        private class ObjectToCustomerConverter : JsonConverter<Customer>
+        {
+            public override bool CanConvert(Type typeToConvert)
+            {
+                return (typeToConvert == typeof(Customer) || typeToConvert == typeof(DerivedCustomer));
+            }
+
+            public override Customer Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                Customer customer = null;
+                if (typeToConvert == typeof(Customer))
+                {
+                    customer = new Customer();
+                }
+
+                if (typeToConvert == typeof(DerivedCustomer))
+                {
+                    customer = new DerivedCustomer();
+                }
+
+                if (customer != null)
+                { 
+                    while (reader.Read())
+                    {
+                        if (reader.TokenType == JsonTokenType.EndObject)
+                        {
+                            return customer;
+                        }
+
+                        if (reader.TokenType == JsonTokenType.PropertyName)
+                        {
+                            string propertyName = reader.GetString();
+                            reader.Read();
+                            switch (propertyName)
+                            {
+                                case "CreditLimit":
+                                    decimal creditLimit = reader.GetDecimal();
+                                    customer.CreditLimit = creditLimit - 1;
+                                    break;
+                                case "Address":
+                                    string city = reader.GetString();
+                                    customer.Address.City = string.IsNullOrEmpty(city) ? "NA" : city;
+                                    break;
+                                case "Name":
+                                    string name = reader.GetString().ToUpper();
+                                    customer.Name = name;
+                                    break;
+                            }
+                        }
+                    }
+                    return customer;
+                }
+
+                throw new NotSupportedException();
+            }
+
+            public override void Write(Utf8JsonWriter writer, Customer value, JsonSerializerOptions options)
+            {
+                writer.WriteStartObject();
+                writer.WriteString("Name", value.Name);
+                writer.WriteNumber("CreditLimit", value.CreditLimit);
+                writer.WriteString("Address", value.Address.City);
+                writer.WriteEndObject();
+            }
+        }
+
+        [Fact]
+        public static void ClassWithFieldHavingCustomConverterTest()
+        {
+            TestClassWithFieldHavingCustomConverter testObject = new TestClassWithFieldHavingCustomConverter { IntValue = 32,
```

Comment:
nit: perhaps a new line for each property setter to make it a bit more readable?